tests/e2escenarios/e2e-test.go

```diff
@@ -0,0 +1,258 @@
+			Expect(err).ToNot(HaveOccurred())
+			defer resp.Body.Close()
+
+			body, _ := io.ReadAll(resp.Body)
+			return string(body)
+		}, 60*time.Second, 10*time.Second).Should(Equal(assertString))
+	}
+
+	Context("starting with empty Directory", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			helper.Chdir(commonVar.Context)
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(BeEmpty())
+		})
+
+		It("should verify developer workflow from empty Directory", func() {
+			deploymentName := "my-component"
+			serviceName := "my-cs"
+			getDeployArgs := []string{"get", "deployment", "-n", commonVar.Project}
+			getSVCArgs := []string{"get", "svc", "-n", commonVar.Project}
+
+			command := []string{"odo", "init"}
+			output, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "javascript")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "Node.js\n")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "nodejs-starter\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(output).To(ContainSubstring("Your new component '" + componentName + "' is ready in the current directory"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("server.js"))
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var outContents []byte
+			var ports map[string]string
+
+			devSession, outContents, _, ports, err = helper.StartDevMode()
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js", "from updated Node.js")
+			Expect(err).ToNot(HaveOccurred())
+			helper.MatchAllInOutput(string(outContents), []string{"Watching for changes in the current directory"})
+
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from updated Node.js Starter Application!")
+
+			// "changes are made made to the applications"
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from updated Node.js", "from Node.js app v2")
+			helper.MatchAllInOutput(string(outContents), []string{"Watching for changes in the current directory"})
+
+			// "should deploy new changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from Node.js app v2 Starter Application!")
+
+			// "running odo list"
+			stdout := helper.Cmd("odo", "list").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{componentName, "nodejs", "Dev"})
+
+			// "exit dev mode and run odo deploy"
+			devSession.Stop()
+			devSession.WaitEnd()
+
+			// "run odo deploy"
```

Comment:
You could check at this point that there are no more resources in the namespace

Reply:
By "resources", do you mean deployment ? 
or how do we do it?